### ISSUE TITLE
Permanent plugin installation with certbot-auto

### ISSUE THREAD

**Issue by adriankirchner** [certbot/certbot#5766]
## My operating system is (include version):
Ubuntu 16.04.1 LTS

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
certbot-auto version 0.22.2

## I ran this command and it produced this output:
```bash
$ certbot-auto renew
Upgrading certbot-auto 0.22.1 to 0.22.2...
Replacing certbot-auto...
Creating virtual environment...
Installing Python packages...
Installation succeeded.

[..]
-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/some-wildcard-certificate.conf
-------------------------------------------------------------------------------
Renewal configuration file /etc/letsencrypt/renewal/some-wildcard-certificate.conf (cert: some-wildcard-certificate) produced an unexpected error: 'Namespace' object has no attribute 'dns_cloudflare_credentials'. Skipping.
[..]
```

## Certbot's behavior differed from what I expected because:
The plugin `certbot-dns-cloudflare` was installed via `/opt/eff.org/certbot/venv/bin/pip install certbot-dns-cloudflare` at version 0.22.1. Our nightly certificate renewal process via `certbot-auto renew` updated certbot and purged the installed plugin. As it seems this is [intended behavior](https://community.letsencrypt.org/t/example-of-using-certbot-auto-with-route-53-dns/53021/8) but I expect a mission critical plugin like that to be installed permanently.

**Comment by ohemorange** [comment]
@bmw, did we ever fix the issue with certbot-auto blowing away the venv on update?

**Comment by bmw** [comment]
No. An attempt was made in #4047, but it didn't work.

I think this issue is covered by #1279 and #4767 so I'm closing this, but if you think I overlooked something, please comment and I'll consider reopening.

**Comment by ExcelNet-Public** [comment]
Funny, both #1279 and #4767 were closed referencing this ticket and you close this referencing them.  This leaves this issue to NEVER be resolved!  Updated to 0.23 and broke the DNS plugins until reinstall again.  This needs an automated fix.
